### PR TITLE
Add unsigned serializer which converts integers to signed, because sq…

### PR DESCRIPTION
**Pull request by egorpugin**
…lite accepts only them. Add test.

### CODE REVIEW

File: tests/IntegralTest.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      id INTEGER PRIMARY KEY AUTOINCREMENT,
+      fp REAL
+  ))");
+
+  // check for automatic conversion in serializer
+  uint64_t v = 17032080461028570721ULL;
+  auto v2 = static_cast<int64_t>(v);
+
+  db(insert_into(fp).set(fp.id = v));
+  db(insert_into(fp).set(fp.id = v2));
```
Comment (rbock) [inline tests/IntegralTest.cpp:83]: I think you are assigning the same value twice to the primary key.

Comment (egorpugin) [reply tests/IntegralTest.cpp:83]: Yes, I'm in process of fixing it. (Before serializer changes, there were different values.)

**Comment by egorpugin** [comment]
Test results

Old code:
```
Sqlite3 debug: Preparing: 'CREATE TABLE fp_sample (
      id INTEGER,
      fp REAL
  )'
Sqlite3 debug: Preparing: 'INSERT INTO fp_sample (id) VALUES(17032080461028570721)'
Sqlite3 debug: Preparing: 'INSERT INTO fp_sample (id) VALUES(-1414663612680980895)'
Sqlite3 debug: Preparing: 'INSERT INTO fp_sample (id) VALUES(?1)'
Sqlite3 debug: Constructing prepared_statement, using handle at 0000019500582EC0
Sqlite3 debug: reseting prepared statement
Sqlite3 debug: binding integral parameter -1414663612680980895 at index: 0, being not null
Sqlite3 debug: reseting prepared statement
Sqlite3 debug: binding integral parameter -1414663612680980895 at index: 0, being not null
Sqlite3 debug: Preparing: 'SELECT fp_sample.id FROM fp_sample'
Sqlite3 debug: Constructing bind result, using handle at 0000019500582E20
Sqlite3 debug: Accessing next row of handle at 0000019500582E20
Sqlite3 debug: binding integral result 0 at index: 0
99: 9223372036854775807 != 17032080461028570721
```

With this patch:
```
Sqlite3 debug: Preparing: 'CREATE TABLE fp_sample (
      id INTEGER,
      fp REAL
  )'
Sqlite3 debug: Preparing: 'INSERT INTO fp_sample (id) VALUES(-1414663612680980895)'
Sqlite3 debug: Preparing: 'INSERT INTO fp_sample (id) VALUES(-1414663612680980895)'
Sqlite3 debug: Preparing: 'INSERT INTO fp_sample (id) VALUES(?1)'
Sqlite3 debug: Constructing prepared_statement, using handle at 00000158E49E2260
Sqlite3 debug: reseting prepared statement
Sqlite3 debug: binding integral parameter -1414663612680980895 at index: 0, being not null
Sqlite3 debug: reseting prepared statement
Sqlite3 debug: binding integral parameter -1414663612680980895 at index: 0, being not null
Sqlite3 debug: Preparing: 'SELECT fp_sample.id FROM fp_sample'
Sqlite3 debug: Constructing bind result, using handle at 00000158E49E2350
Sqlite3 debug: Accessing next row of handle at 00000158E49E2350
Sqlite3 debug: binding integral result 0 at index: 0
Sqlite3 debug: Accessing next row of handle at 00000158E49E2350
Sqlite3 debug: binding integral result -1414663612680980895 at index: 0
Sqlite3 debug: Accessing next row of handle at 00000158E49E2350
Sqlite3 debug: binding integral result -1414663612680980895 at index: 0
Sqlite3 debug: Accessing next row of handle at 00000158E49E2350
Sqlite3 debug: binding integral result -1414663612680980895 at index: 0
Sqlite3 debug: Accessing next row of handle at 00000158E49E2350
```

**Comment by rbock** [comment]
Looks good to me. Thanks for taking care of the issue!

**Pull request merged**